packages/node/src/utils/near.ts

```diff
@@ -244,12 +335,54 @@ export function filterTransactions(
   );
 }
 
+export function filterReceipt(
+  receipt: NearTransactionReceipt,
+  filter?: NearReceiptFilter,
+): boolean {
+  if (!filter) return true;
+  if (filter.sender && receipt.predecessor_id !== filter.sender) return false;
```

Comment:
Is the predecessor id the sender or the contract that emitted this receipt?

Reply:
The account that emitted the receipt. signer id is the one that signed off the transaction